### ISSUE TITLE
SteamVR-OSVR always tells OpenVR to apply a head model

### ISSUE THREAD

**Issue by 0x1100** [OSVR/SteamVR-OSVR#105]
## It should not

The driver [always](https://github.com/OSVR/SteamVR-OSVR/blob/21ed38d76aee15d963641c40afbf6d8da8d1f00f/src/OSVRTrackedDevice.cpp#L885) tells OpenVR to apply a head model when supplying the tracked position.
- When using the camera, the additional head model results in a wrong tracking.
- When using the camera, relying on OpenVR's head model implies we don't get one in OSVR apps/demos like the sample scene from the tray app.

## Proposed solution

A head model can easily be applied in the OSVR server configuration file. We should never ask OpenVR to add one and, instead, do it ourselves in OSVR if we're not using the camera. This way, people would get the same result whether they're using a Steam app or an OSVR one.

```diff
diff --git a/src/OSVRTrackedDevice.cpp b/src/OSVRTrackedDevice.cpp
index b450f24..cfee739 100644
--- a/src/OSVRTrackedDevice.cpp
+++ b/src/OSVRTrackedDevice.cpp
@@ -882,7 +882,7 @@ void OSVRTrackedDevice::HmdTrackerCallback(void* userdata, const OSVR_TimeValue*
     pose.result = vr::TrackingResult_Running_OK;
     pose.poseIsValid = true;
     pose.willDriftInYaw = true;
-    pose.shouldApplyHeadModel = true;
+    pose.shouldApplyHeadModel = false;
     pose.deviceIsConnected = true;
 
     self->pose_ = pose;
```

The following alias adds a head model close to the one supplied by OpenVR.
A similar change will be done to the no-cam json files supplied by the AIO release.

```json
"aliases": {
    "/headSpace": {
        "translate": [0.0, 0.10, -0.07],
        "child": "/com_osvr_Multiserver/OSVRHackerDevKit0/semantic/hmd"
    },
    "/me/head": "/headSpace"
}
```

**Comment by godbyk** [comment]
Since this entails modifying the configuration files in the OSVR-Core repository, I would suggest posting a PR there first. Once that's been merged in, I can disable the SteamVR head model.

**Comment by 0x1100** [comment]
No. It does not entails modifying the configuration files in the OSVR-Core repository since those that are complete all use the camera.

In any case, two wrongs don't make a right and both need to be fixed for the next release anyway so there is no good reason to make it drag on.

**Comment by godbyk** [comment]
@0x1100 In your original post, you said:

> A head model can easily be applied in the OSVR server configuration file.

So to clarify, is a head model is already applied in the OSVR server configuration files?

> In any case, two wrongs don't make a right and both need to be fixed for the next release anyway so there is no good reason to make it drag on.

I'm trying to ensure the head model doesn't get removed entirely. I don't care whether it's applied on the OSVR server side or on the SteamVR side, but it must be one or the other, not both or neither.

**Comment by rpavlik** [comment]
There's not a head model in the default configurations in OSVR-Core nor in any config with just an IMU, though as you noted it is easy to add one.

The presence of such a transform will result in position data being reported for the interface (since rotations will result in translation as well). Thus, I suggest detecting whether or not there is such a configuration by opening an object for /me/head, waiting until it has orientation state, then checking to see if it has position state. If it does have position state, then don't apply a SteamVR head model.

**Comment by 0x1100** [comment]
@rpavlik's solution is better but still sounds easy to implement.
How fast can you do it?
